ci/ssm.py

```diff
@@ -0,0 +1,253 @@
+            value = var['value']
+            var_type = var['var_type']
+            if verbose:
+                print("Adding parameter:")
+                print("     Name: {}".format(name))
+                print("     Value: {}".format(value))
+                print("     Type: {}".format(var_type))
+                print("\n")
+            put_parameter(name, value, var_type)
+
+
+def get_parameters(client, path, **kwargs):
+    """Get SSM parameters."""
+    next_token = kwargs.get('next_token', None)
+    verbose = kwargs.get('verbose', None)
+    if next_token:
+        response = client.get_parameters_by_path(
+            Path=path,
+            Recursive=True,
+            WithDecryption=True,
+            MaxResults=10,
+            NextToken=next_token
+        )
+        if verbose:
+            print("Recieved NextToken with response: {}".format(response))
+    else:
+        response = client.get_parameters_by_path(
+            Path=path,
+            Recursive=True,
+            WithDecryption=True,
+            MaxResults=10
+        )
+        if verbose:
+            print("Recieved response: {}".format(response))
+    return response
+
+
+def get_env_paths():
+    """Get environment variable (ssm) paths."""
+    environment_paths = []
+    zones = [
+        "commercial",
+        "govcloud"
+    ]
+    project_types = [
+        "ci",
+        "rc",
+        "official"
+    ]
+    for type in project_types:
+        for zone in zones:
+            environment_paths.append("/spel/{}_{}".format(zone, type))
+    return environment_paths
+
+
+def get_ssm_path(env):
+    """Return the ssm path for an environment."""
+    return "/spel/{}/".format(env)
+
+
+def get_parameters_by_path(path):
+    """Get all parameters."""
+    parameters = []
+    client = boto3.client('ssm')
+    response = get_parameters(client, path)
+    parameters += response['Parameters']
+    try:
+        while response['NextToken']:
+            response = get_parameters(client, path,
+                                      next_token=response['NextToken'])
+            parameters += response['Parameters']
+    except KeyError:
+        return parameters
+
+
+def get_all_remote_parameters():
+    """Get all variable values."""
+    parameters = []
+    paths = get_env_paths()
+    for path in paths:
+        parameters += get_parameters_by_path(path)
+    return parameters
+
+
+def compare_ssm_to_local(verbose=None, target_env=None):
+    """Compare variables.yaml file to SSM."""
+    ssm_values = get_all_remote_parameters()
+    local_values = load_local_parameters()
+    # targeted environment search
+    if target_env:
+        local_values = local_values[target_env]
+        ssm_path = "/spel/{}/".format(target_env)
+        for local_variable in local_values:
+            for ssm_variable in ssm_values:
+                ssm_var_name = ssm_variable['Name'].replace(ssm_path, '')
+                if '/spel/' not in ssm_var_name:
```

Comment:
These nested if statements can be combined (easy to spot because the else blocks are empty)...

```
if (
    '/spel/' not in ssm_var_name and 
    local_variable['var_name'] == ssm_var_name
):
```

Reply:
consolidated